OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs

```diff
@@ -103,6 +103,24 @@ public MapEditorLogic(Widget widget, World world, WorldRenderer worldRenderer)
 				if (reslayer != null)
 					cashLabel.GetText = () => "$ {0}".F(reslayer.NetWorth);
 			}
+
+			var searchTextField = widget.GetOrNull<TextFieldWidget>("MAPSEARCH_TEXTFIELD");
+			if (searchTextField != null)
+			{
+				searchTextField.OnEnterKey = () =>
+				{
+					var yamlNode = world.Map.ActorDefinitions.FirstOrDefault(ad => ad.Key == "Actor" + searchTextField.Text);
```

Comment:
You can't assume that all actor names start with "Actor".  This is the convention for map placed actors, but won't be true for scripted maps.